Userland/Libraries/LibGUI/TextEditor.cpp

```diff
@@ -1179,14 +1179,8 @@ void TextEditor::timer_event(Core::TimerEvent&)
         update_cursor();
 }
 
-bool TextEditor::write_to_file(const String& path)
+bool TextEditor::write_to_file(int fd)
 {
-    int fd = open(path.characters(), O_WRONLY | O_CREAT | O_TRUNC, 0666);
-    if (fd < 0) {
-        perror("open");
-        return false;
-    }
-
     ScopeGuard fd_guard = [fd] { close(fd); };
```

Comment:
Since you no longer `open` the file descriptor in this message, it should probably also not be closed here, which is what the `ScopeGuard` does. It normally makes sense for the code that opens the file descriptor to also close it. Otherwise, it would make sense to declare in the method name that you will close the file descriptor (e.g. `write_to_file_and_close_fd`).

Reply:
Taking the latter path.